src/AutoRest.CSharp/Common/Output/Models/Types/HelperTypeProviders/MultipartFormDataRequestContentProvider.cs

```diff
@@ -0,0 +1,495 @@
+            var contentTypeParam = new Parameter("contentType", null, new CSharpType(typeof(string), true),
+                        Constant.Default(new CSharpType(typeof(string), true)), ValidationType.None, null);
+            var signature = new MethodSignature(
+                Name: _addMethodName,
+                Summary: null,
+                Description: null,
+                Modifiers: MethodSignatureModifiers.Public,
+                ReturnType: null,
+                ReturnDescription: null,
+                Parameters: new[]
+                {
+                    contentParam,
+                    nameParam,
+                    filenameParam,
+                    contentTypeParam
+                });
+            MethodBodyStatement? valueDelareStatement = null;
+            ValueExpression contentExpression;
+            contentExpression = New.Instance(typeof(ByteArrayContent), contentParam);
+            switch (typeof(T))
+            {
+                case Type type when type == typeof(Stream):
+                    contentExpression = New.Instance(typeof(StreamContent), contentParam);
+                    break;
+                case Type type when type == typeof(string):
+                    contentExpression = New.Instance(typeof(StringContent), contentParam);
+                    break;
+                case Type btype when btype == typeof(bool):
+                    var boolToStringValue = new TernaryConditionalOperator(contentParam, Literal("true"), Literal("false"));
+                    valueDelareStatement = Declare(typeof(string), "value", boolToStringValue, out TypedValueExpression boolVariable);
+                    contentExpression = New.Instance(typeof(StringContent), boolVariable);
+                    break;
+                case Type itype when itype == typeof(int):
+                case Type ftype when ftype == typeof(float):
+                case Type ltype when ltype == typeof(long):
+                case Type dtype when dtype == typeof(double):
+                case Type detype when detype == typeof(decimal):
+                    ValueExpression invariantCulture = new MemberExpression(typeof(CultureInfo), nameof(CultureInfo.InvariantCulture));
+                    var invariantCultureValue = new InvokeInstanceMethodExpression(contentParam, nameof(Int32.ToString), new[] {Literal("G"), invariantCulture }, null, false);
+                    valueDelareStatement = Declare(typeof(string), "value", invariantCultureValue, out TypedValueExpression variable);
+                    contentExpression = New.Instance(typeof(StringContent), variable);
+                    break;
+                case Type bdtype when bdtype == typeof(BinaryData):
+                    contentExpression = New.Instance(typeof(ByteArrayContent), new BinaryDataExpression(contentParam).ToArray());
+                    break;
+            }
```

Comment:
well it might be more clear if we make it a if-else if block
Also we need an escaping branch to throw exception so that we could know something is wrong if some incorrect change has been made.